### PR TITLE
[Enhancement][Feature][BugFix] fix show proc '/current_queries'

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this：
- [x] BugFix
- [x] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

In the return result of `show proc '/current_queries'` and `show proc '/current_queries/${query_id}/fragments'`, the io statistics are wrong. In this PR, I fixed it and add cpu cost into the result so that we can get the accurate statistics of running queries.

Major changes:
1. refactor the workflow of `show proc '/current_queries'`. We obtain the query statistics from BE and merge them in FE according to the query_ids.
2. remove `show proc '/current_queries/${query_id}/fragments'` because fragment level stats is not necessary in pipeline engine and add `show proc '/current_queries/${query_id}/hosts'` to show the cost in each related BE server for a specific query.
3. remove some unused code.

**NOTE** 
this feature will only collect queries executing by the pipeline engine.

here is an example, there are three running queries in a cluster with three BEs and one FE.
we can know how many queries running in this FE by `show proc '/current_queries'`, the current stats in each BE of a specific query by `show proc '/current_queries/${query_id}/hosts'` and what the query is by `show proc '/current_queries/${query_id}'`

```sql
mysql> show proc '/current_queries';
+--------------------------------------+--------------+------------------------+------+-----------+----------------+----------------+----------+
| QueryId                              | ConnectionId | Database               | User | ScanBytes | ProcessRows    | CPUCostSeconds | ExecTime |
+--------------------------------------+--------------+------------------------+------+-----------+----------------+----------------+----------+
| 5ea577e0-6ac4-11ed-bef2-00163e14d435 | 148          | tpcds_100g_performance | root | 4.28 GB   | 157046219 Rows | 98.81 Seconds  | 8082     |
| 5ea7c1d1-6ac4-11ed-bef2-00163e14d435 | 149          | tpcds_100g_performance | root | 4.27 GB   | 156542581 Rows | 99.18 Seconds  | 8082     |
| 5ecef8e2-6ac4-11ed-bef2-00163e14d435 | 147          | tpcds_100g_performance | root | 4.18 GB   | 153039530 Rows | 95.20 Seconds  | 7799     |
+--------------------------------------+--------------+------------------------+------+-----------+----------------+----------------+----------+
3 rows in set (0.01 sec)

mysql> show proc '/current_queries/5ea577e0-6ac4-11ed-bef2-00163e14d435/hosts';
+--------------------+-----------+---------------+----------------+
| Host               | ScanBytes | ScanRows      | CpuCostSeconds |
+--------------------+-----------+---------------+----------------+
| 172.26.34.216:8060 | 2.30 GB   | 86045370 Rows | 52.95 Seconds  |
| 172.26.34.217:8060 | 2.35 GB   | 88019928 Rows | 53.35 Seconds  |
| 172.26.34.218:8060 | 2.35 GB   | 87395083 Rows | 52.63 Seconds  |
+--------------------+-----------+---------------+----------------+
3 rows in set (0.01 sec)

mysql> show proc '/current_queries/5ea577e0-6ac4-11ed-bef2-00163e14d435';
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Sql                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| with year_total as ( select c_customer_id customer_id ,c_first_name customer_first_name ,c_last_name customer_last_name ,c_preferred_cust_flag customer_preferred_cust_flag ,c_birth_country customer_birth_country ,c_login customer_login ,c_email_address customer_email_address ,d_year dyear ,sum(ss_ext_list_price-ss_ext_discount_amt) year_total ,'s' sale_type from customer ,store_sales ,date_dim where c_customer_sk = ss_customer_sk and ss_sold_date_sk = d_date_sk group by c_customer_id ,c_first_name ,c_last_name ,c_preferred_cust_flag ,c_birth_country ,c_login ,c_email_address ,d_year union all select c_customer_id customer_id ,c_first_name customer_first_name ,c_last_name customer_last_name ,c_preferred_cust_flag customer_preferred_cust_flag ,c_birth_country customer_birth_country ,c_login customer_login ,c_email_address customer_email_address ,d_year dyear ,sum(ws_ext_list_price-ws_ext_discount_amt) year_total ,'w' sale_type from customer ,web_sales ,date_dim where c_customer_sk = ws_bill_customer_sk and ws_sold_date_sk = d_date_sk group by c_customer_id ,c_first_name ,c_last_name ,c_preferred_cust_flag ,c_birth_country ,c_login ,c_email_address ,d_year ) select t_s_secyear.customer_id ,t_s_secyear.customer_first_name ,t_s_secyear.customer_last_name ,t_s_secyear.customer_preferred_cust_flag from year_total t_s_firstyear ,year_total t_s_secyear ,year_total t_w_firstyear ,year_total t_w_secyear where t_s_secyear.customer_id = t_s_firstyear.customer_id and t_s_firstyear.customer_id = t_w_secyear.customer_id and t_s_firstyear.customer_id = t_w_firstyear.customer_id and t_s_firstyear.sale_type = 's' and t_w_firstyear.sale_type = 'w' and t_s_secyear.sale_type = 's' and t_w_secyear.sale_type = 'w' and t_s_firstyear.dyear = 2001 and t_s_secyear.dyear = 2001+1 and t_w_firstyear.dyear = 2001 and t_w_secyear.dyear = 2001+1 and t_s_firstyear.year_total > 0 and t_w_firstyear.year_total > 0 and case when t_w_firstyear.year_total > 0 then t_w_secyear.year_total / t_w_firstyear.year_total else 0.0 end > case when t_s_firstyear.year_total > 0 then t_s_secyear.year_total / t_s_firstyear.year_total else 0.0 end order by t_s_secyear.customer_id ,t_s_secyear.customer_first_name ,t_s_secyear.customer_last_name ,t_s_secyear.customer_preferred_cust_flag limit 100;  |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 115 (02.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|6|00.00%|[189, 190, 191, 192, 193, 194]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|81|00.00%|[100, 236, 237, 238, 239, 240, 241, 243, 244, 245, 246, 247, 248, 250, 251, 252, 253, 254, 256, 257, 258, 259, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 278, 279, 280, 281, 282, 283, 284, 306, 307, 309, 310, 311, 313, 314, 315, 316, 317, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 374, 375, 376, 378, 380, 383, 387, 388, 391, 395, 396, 399, 403, 404, 488, 504, 508, 509]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|9|22.22%|[77, 78, 80, 81, 82, 92, 93]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|1|3|33.33%|[105, 108]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 219 (01.37%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|9|00.00%|[47, 78, 79, 85, 86, 87, 91, 92, 100]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectFragmentStatisticsRequest.java|0|6|00.00%|[21, 23, 25, 26, 27, 28]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|12|00.00%|[190, 191, 192, 193, 194, 195, 202, 203, 204, 205, 206, 207]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|164|00.00%|[103, 133, 243, 244, 246, 247, 248, 250, 251, 252, 253, 254, 255, 257, 258, 259, 260, 261, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 277, 278, 279, 281, 282, 283, 288, 289, 290, 291, 292, 293, 294, 295, 296, 298, 299, 300, 301, 302, 303, 304, 309, 310, 312, 313, 314, 315, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 334, 335, 336, 337, 338, 339, 341, 342, 343, 344, 345, 346, 348, 349, 350, 351, 352, 354, 355, 356, 357, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 376, 377, 378, 379, 380, 381, 382, 404, 405, 407, 408, 409, 411, 412, 413, 414, 415, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 472, 473, 474, 476, 478, 481, 485, 486, 489, 493, 494, 497, 501, 502, 547, 548, 549, 550, 551, 552, 555, 559, 563, 567, 571, 575, 625, 641, 645, 646]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|9|22.22%|[77, 78, 80, 81, 82, 92, 93]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|1|3|33.33%|[105, 108]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 165 (01.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|7|00.00%|[47, 85, 86, 87, 91, 92, 100]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectFragmentStatisticsRequest.java|0|6|00.00%|[21, 23, 25, 26, 27, 28]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|12|00.00%|[190, 191, 192, 193, 194, 195, 202, 203, 204, 205, 206, 207]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|114|00.00%|[67, 71, 76, 77, 79, 80, 81, 83, 84, 85, 86, 87, 88, 90, 91, 92, 93, 94, 96, 97, 98, 99, 101, 102, 103, 104, 105, 106, 107, 108, 110, 111, 112, 114, 116, 121, 122, 123, 124, 125, 126, 127, 128, 129, 131, 132, 133, 134, 135, 136, 137, 142, 143, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 159, 183, 186, 191, 194, 195, 196, 197, 198, 199, 200, 201, 203, 204, 206, 214, 215, 217, 218, 219, 220, 221, 222, 223, 225, 226, 227, 230, 231, 234, 255, 256, 257, 259, 264, 268, 276, 277, 280, 284, 295, 299, 319, 323, 335, 351, 355, 356]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|7|28.57%|[80, 81, 82, 92, 93]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|1|3|33.33%|[105, 108]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 165 (01.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|7|00.00%|[47, 85, 86, 87, 91, 92, 100]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectFragmentStatisticsRequest.java|0|6|00.00%|[21, 23, 25, 26, 27, 28]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|12|00.00%|[190, 191, 192, 193, 194, 195, 202, 203, 204, 205, 206, 207]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|114|00.00%|[67, 71, 76, 77, 79, 80, 81, 83, 84, 85, 86, 87, 88, 90, 91, 92, 93, 94, 96, 97, 98, 99, 101, 102, 103, 104, 105, 106, 107, 108, 110, 111, 112, 114, 116, 121, 122, 123, 124, 125, 126, 127, 128, 129, 131, 132, 133, 134, 135, 136, 137, 142, 143, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 159, 183, 186, 191, 194, 195, 196, 197, 198, 199, 200, 201, 203, 204, 206, 214, 215, 217, 218, 219, 220, 221, 222, 223, 225, 226, 227, 230, 231, 234, 255, 256, 257, 259, 264, 268, 276, 277, 280, 284, 295, 299, 319, 323, 335, 351, 355, 356]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|7|28.57%|[80, 81, 82, 92, 93]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|1|3|33.33%|[105, 108]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 165 (01.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|7|00.00%|[47, 85, 86, 87, 91, 92, 100]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectFragmentStatisticsRequest.java|0|6|00.00%|[21, 23, 25, 26, 27, 28]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|12|00.00%|[190, 191, 192, 193, 194, 195, 202, 203, 204, 205, 206, 207]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|114|00.00%|[67, 71, 76, 77, 79, 80, 81, 83, 84, 85, 86, 87, 88, 90, 91, 92, 93, 94, 96, 97, 98, 99, 101, 102, 103, 104, 105, 106, 107, 108, 110, 111, 112, 114, 116, 121, 122, 123, 124, 125, 126, 127, 128, 129, 131, 132, 133, 134, 135, 136, 137, 142, 143, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 159, 183, 186, 191, 194, 195, 196, 197, 198, 199, 200, 201, 203, 204, 206, 214, 215, 217, 218, 219, 220, 221, 222, 223, 225, 226, 227, 230, 231, 234, 255, 256, 257, 259, 264, 268, 276, 277, 280, 284, 295, 299, 319, 323, 335, 351, 355, 356]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|7|28.57%|[80, 81, 82, 92, 93]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|1|3|33.33%|[105, 108]|

File: be/src/exec/pipeline/fragment_context.h [hunk 1]
```diff
@@ -164,6 +174,10 @@ class FragmentContext {
     PerDriverScanRangesMap _scan_ranges_per_driver_seq;
     std::vector<StreamLoadContext*> _stream_load_contexts;
     bool _channel_stream_load = false;
+
+    std::atomic<int64_t> _total_cpu_cost_ns = 0;
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply be/src/exec/pipeline/fragment_context.h:178]: > do really need the fragment level statistics?

I think it's less important than query level stats, but this information can help us to identify which backend nodes are affected by a specific query. 
And `show proc '/current_queries/${query_id}/fragments'` is an existing grammar in our system, I tend to keep it first.If the collection of fragment level stats will affect performance, we can consider removing it.

File: be/src/service/internal_service.cpp [hunk 2]
```diff
@@ -403,6 +403,25 @@ void PInternalServiceImplBase<T>::trigger_profile_report(google::protobuf::RpcCo
...
+    _exec_env->query_context_mgr()->collect_query_statistics(request, result);
+}
+
+template <typename T>
+void PInternalServiceImplBase<T>::collect_fragment_statistics(google::protobuf::RpcController* controller,
+                                                              const PCollectFragmentStatisticsRequest* request,
+                                                              PCollectFragmentStatisticsResult* result,
+                                                              google::protobuf::Closure* done) {
+    ClosureGuard closure_guard(done);
+    LOG(INFO) << "collect fragment statistics";
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply be/src/service/internal_service.cpp]: > remove it?

fixed

**Comment by silverbullet233** [comment]
run starrocks_admit_test

File: gensrc/proto/internal_service.proto [hunk 3]
```diff
@@ -291,6 +291,38 @@ message PTriggerProfileReportResult {
     required StatusPB status = 1;
 };
 
+message PCollectQueryStatisticsRequest {
+    repeated PUniqueId query_ids = 1;
```
Comment (ZiheLiu) [inline gensrc/proto/internal_service.proto:295]: All the `repeated` fields should also be labeled `optional`?

File: fe/fe-core/src/main/java/com/starrocks/qe/QeProcessorImpl.java [hunk 4]
```diff
@@ -134,6 +136,7 @@ public TReportExecStatusResult reportExecStatus(TReportExecStatusParams params,
             return result;
         }
         try {
+            LOG.info("update fragment exec status");
```
[the remark this answers is not in the document]
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/qe/QeProcessorImpl.java]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 164 (01.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryFragmentProcNode.java|0|7|00.00%|[47, 85, 86, 87, 91, 92, 100]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|2|00.00%|[105, 108]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 44]|
|:large_blue_circle:|com/starrocks/rpc/PCollectFragmentStatisticsRequest.java|0|6|00.00%|[21, 23, 25, 26, 27, 28]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|12|00.00%|[190, 191, 192, 193, 194, 195, 202, 203, 204, 205, 206, 207]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|114|00.00%|[67, 71, 76, 77, 79, 80, 81, 83, 84, 85, 86, 87, 88, 90, 91, 92, 93, 94, 96, 97, 98, 99, 101, 102, 103, 104, 105, 106, 107, 108, 110, 111, 112, 114, 116, 121, 122, 123, 124, 125, 126, 127, 128, 129, 131, 132, 133, 134, 135, 136, 137, 142, 143, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 159, 183, 186, 191, 194, 195, 196, 197, 198, 199, 200, 201, 203, 204, 206, 214, 215, 217, 218, 219, 220, 221, 222, 223, 225, 226, 227, 230, 231, 234, 255, 256, 257, 259, 264, 268, 276, 277, 280, 284, 295, 299, 319, 323, 335, 351, 355, 356]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|7|28.57%|[80, 81, 82, 92, 93]|

**Review by murphyatwork — DISMISSED** [review]

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 145 (01.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|1|00.00%|[107]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQuerySqlProcDir.java|0|3|00.00%|[57, 58, 60]|
|:large_blue_circle:|com/starrocks/common/util/QueryStatisticsFormatter.java|0|4|00.00%|[41, 42, 43, 45]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryHostProcNode.java|0|26|00.00%|[18, 19, 20, 24, 25, 26, 30, 31, 32, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 48, 51, 52, 53, 54]|
|:large_blue_circle:|com/starrocks/rpc/PCollectQueryStatisticsRequest.java|0|6|00.00%|[19, 20, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|0|6|00.00%|[189, 190, 191, 192, 193, 194]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|86|00.00%|[65, 69, 70, 72, 73, 75, 76, 77, 78, 79, 80, 82, 83, 84, 85, 87, 89, 94, 95, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 110, 111, 112, 113, 114, 115, 116, 118, 142, 145, 150, 153, 154, 155, 156, 157, 158, 159, 160, 162, 163, 165, 173, 174, 176, 177, 178, 179, 180, 181, 182, 183, 185, 186, 187, 190, 191, 192, 193, 194, 195, 198, 219, 220, 221, 223, 228, 232, 236, 240, 244, 248, 258, 266, 270]|
|:large_blue_circle:|com/starrocks/qe/QueryStatisticsItem.java|0|6|00.00%|[49, 86, 145, 146, 179, 180]|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryStatisticsProcDir.java|2|7|28.57%|[80, 81, 82, 92, 93]|

**Review by murphyatwork — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by silverbullet233** [comment]
run starrocks_clang

**Comment by silverbullet233** [comment]
run starrocks_clang-tidy

**Comment by mofeiatwork** [comment]
run starrocks_be_unittest

**Pull request merged**